src/main/fc/rc.c

```diff
@@ -395,16 +394,12 @@ uint16_t getCurrentRxRefreshRate(void)
 #ifdef USE_RC_SMOOTHING_FILTER
 // Determine a cutoff frequency based on filter type and the calculated
 // average rx frame time
-FAST_CODE_NOINLINE int calcRcSmoothingCutoff(int avgRxFrameTimeUs, bool pt1, uint8_t autoSmoothnessFactor)
+FAST_CODE_NOINLINE int calcRcSmoothingCutoff(int avgRxFrameTimeUs, uint8_t autoSmoothnessFactor)
 {
     if (avgRxFrameTimeUs > 0) {
-        const float cutoffFactor = (100 - autoSmoothnessFactor) / 100.0f;
-        float cutoff = (1 / (avgRxFrameTimeUs * 1e-6f)) / 2;  // calculate the nyquist frequency
+        const float cutoffFactor = 1.5f / (1.0f + (autoSmoothnessFactor / 10.0f));
+        float cutoff = (1 / (avgRxFrameTimeUs * 1e-6f));  // link frequency
```

Comment:
Why are we using the link sample frequency and not the nyquist frequency for the link as the cutoff here?

Reply:
I checked the maths used by our PT1 type filters.  
It's been done so there is no issues with requesting a cutoff above Nyquist.
The maths compute the  RC value (RC = 1 / ( 3.10755f * M_PIf * f_cut)); hence RC approximates zero as cutoff increases.
The Filters->k constant k is dT/(RC+dT).  So as RC approaches zero, k approaches 1.
When k approaches 1, input gets closer to 1.  So long as k <=1, the filter is stable.
So the maths we use is unconditionally stable, even if a cutoff was requested at a frequency above that of the loop time, we would not have a problem.

Reply:
The RC smoothing filter is running at pid loop frequency so if it's a nyquist frequency we should care about it would be the one of the pid loop?

Reply:
No, check the maths.  We could set the cutoff to 1,000,000hz, it won't matter.

Reply:
I had to think about this for RC links of 1000hz and a loop time of 2k.  
The maths I use can set the cutoff to 1.5*linkFrequency when auto smoothing is 0.   
For a 2k PID loop that would be above Nyquist.  
But we don't need to restrict the cutoff because the maths we use isn't the standard maths for calculating k.  Our filters of this kind cannot become unstable even if we enter a value above Nyquist.

Reply:
Here is a graph of the actual result when auto smoothing is 0, PT3 selected, loop time 2k, link speed 1k.
In this example, Nyquist is 1000hz, and the calculated cutoff is 1500hz.
This shows the result using our maths, with a series of steps 5ms wide.  I put one step 1ms wide in the middle.
Every successive PID loop is being calculated using our exact maths.
![AboveNyquist](https://user-images.githubusercontent.com/11737748/115996914-8710f100-a624-11eb-80bd-1ac28f302295.jpg)

Reply:
Yes, I'm aware that the PT1, 2, 3 are well behaved and the biquad is not. If the same discretization method had been used for the biquad as for PT1 it would also be well behaved. But it wouldn't be a biquad anymore 😄 
What I'm saying is that if we should be concerned about the nyquist frequency it should be that of the pid loop and not of the update frequency of the rc link. We're feeding the rc signal to the filter at pid loop frequency.  
But yes. I agree with what you're saying about the PT filters. It doesn't really matter since it won't oscillate out of control like the biquad will. I have tested this previously and have seen what happens

Reply:
Oh yeah would be super ugly if k > 1.
The corollary of the maths we use to avoid this, in our PTn filters, is:
- no matter how high we push cutoff, the actual cutoff value never exceeds Nyquist
- as we push cutoff closer to Nyquist, the actual cutoff is less than we intended.
Both these things are 'good' :-)

Reply:
Ok, what you say makes sense @ctzsnooze @klutvott123. But if the cutoff frequency does not matter in our calculation, why are we making it dependent on the link frame frequency? Ideally if we could set it to a constant value we could save a bunch of calculation.